src/oatpp/parser/json/Utils.cpp

```diff
@@ -476,6 +476,15 @@ std::string Utils::parseStringToStdString(ParsingCaret& caret){
   return "";
   
 }
-    
+
+bool Utils::findDecimalSeparatorInNextWord(const char* data) {
+  while (*data != '\0' && std::isspace(*data) == 0) {
```

Comment:
This condition should be changed. 
It checks for `\0` char or until `std::isspace == false` which is incorrect.
Consider the following JSON:

```json
{"a":1,"b":1.1}
```

in this case this method will be evaluated to true when checking `a` since it won't stop on the `,`.

Reply:
Good point. Is there any helper or definition which characters can mark the end of a "value field"? Or should I introduce something globally for the "," and "}"? Or simply use these locally in this method?

Reply:
I think we can do the following check - if the current char is anything but a digit or a decimal separator - stop.

> Or should I introduce something globally for the "," and "}"? Or simply use these locally in this method?

Locally should be enough

Reply:
Thank you for your inputs. I changed the implementation accordingly and tried to use `Caret`. I hope I did understand the usage of `Caret` and `StateSaveGuard` correctly. I also improved the unit tests for numbers a bit to have the case tested where additional content comes after the parsed number.